packages/sit-onyx/src/components/OnyxAccordion/OnyxAccordion.ct.tsx

```diff
@@ -0,0 +1,99 @@
+
+test.describe("ScreenshotTest", () => {
+  executeMatrixScreenshotTest({
+    name: "Accordion",
+    columns: DENSITIES,
+    rows: ["default", "open", "hover", "focus-visible", "disabled", "skeleton"],
+    component: (column, row) => (
+      <OnyxAccordion
+        style="width: 20rem"
+        density={column}
+        skeleton={row === "skeleton"}
+        disabled={row === "disabled"}
+      >
+        <OnyxAccordionItem>
+          <template v-slot:header>Accordion Header 1</template>
+          Accordion Panel 1
+        </OnyxAccordionItem>
+        <OnyxAccordionItem>
+          <template v-slot:header>Accordion Header 2</template>
+          Accordion Panel 2
+        </OnyxAccordionItem>
+      </OnyxAccordion>
+    ),
+    hooks: {
+      beforeEach: async (component, page, _column, row) => {
+        if (row === "open") {
+          await component.evaluate((element) => {
+            element.style.padding = `auto 5rem`;
+          });
+          component.getByRole("button", { name: "Accordion Header 1" }).click();
+        }
+        if (row === "hover") component.getByRole("button", { name: "Accordion Header 1" }).hover();
+        if (row === "focus-visible") await page.keyboard.press("Tab");
+      },
+    },
+  });
+  executeMatrixScreenshotTest({
+    name: "Accordion (disabled)",
+    columns: ["closed", "open"],
+    rows: ["default", "hover", "focus-visible"],
+    component: (columns) => (
+      <OnyxAccordion style="width: 16rem;" disabled>
+        <OnyxAccordionItem open={columns === "open"}>
+          <template v-slot:header>Accordion Header 1</template>
+          Accordion Panel 1
+        </OnyxAccordionItem>
+        <OnyxAccordionItem>
+          <template v-slot:header>Accordion Header 2</template>
+          Accordion Panel 2
+        </OnyxAccordionItem>
+      </OnyxAccordion>
+    ),
+    hooks: {
+      beforeEach: async (component, page, _column, row) => {
+        if (row === "hover")
+          await component.getByRole("button", { name: "Accordion Header 1" }).hover();
+        if (row === "focus-visible") await page.keyboard.press("Tab");
+      },
+    },
+  });
+});
+
+test("should open only one item at a time in exclusive mode", async ({ mount, makeAxeBuilder }) => {
+  const component = await mount(
+    <OnyxAccordion exclusive>
+      <OnyxAccordionItem>
+        <template v-slot:header>Accordion Header 1</template>
+        Accordion Panel 1
+      </OnyxAccordionItem>
+      <OnyxAccordionItem>
+        <template v-slot:header>Accordion Header 2</template>
+        Accordion Panel 2
+      </OnyxAccordionItem>
+    </OnyxAccordion>,
+  );
+
+  const firstHeader = component.locator(".onyx-accordion-item__header").first();
+  const secondHeader = component.locator(".onyx-accordion-item__header").nth(1);
+  const firstPanel = component.locator(".onyx-accordion-item__panel").first();
+  const secondPanel = component.locator(".onyx-accordion-item__panel").nth(1);
```

Comment:
```suggestion
  const firstHeader = component.getByRole('button', { name: 'Accordion Header 1' });
  const secondHeader = component.getByRole('button', { name: 'Accordion Header 2' });
  const firstPanel = component.getByLabel('Accordion Header 1');
  const secondPanel = component.getByLabel('Accordion Header 2');
```

We should try to avoid using `.locator()` and instead using Playwright web native selectors like `getByRole()` or `getByLabel()`.

You can use the VS Code Playwright extension to just click on elements and it will copy the selector :)

<img width="196" alt="image" src="https://github.com/user-attachments/assets/fdf058d9-3d4d-413c-b4f5-e2f5d203ab20" />

![image](https://github.com/user-attachments/assets/092001ca-4a62-4f74-974c-fcae4bbce2c7)

Reply:
Same for the OnyxAccordionItem tests